src/Decimal.php

```diff
@@ -817,6 +819,11 @@ protected function fromScientific(string $value, ?int $scale): void
         $this->negative = $matches[1] === '-';
         /** @var string $value */
```

Comment:
This already makes the check below not needed IMO.
It cannot happen here anyway.

Reply:
okay, I will remove the check. thanks.